apps/nextjs/src/app/additional-material-spike/LessonFinder.tsx

```diff
@@ -0,0 +1,90 @@
+"use client";
+
+import { useEffect, useState } from "react";
+
+import type { OakOpenApiSearchSchema } from "@oakai/additional-materials/src/documents/schemas/oakOpenApi";
+
+import { OakFlex, OakHeading, OakTextInput } from "@oaknational/oak-components";
+import Link from "next/link";
+import { redirect } from "next/navigation";
+
+import { useClientSideFeatureFlag } from "@/components/ContextProviders/FeatureFlagProvider";
+import { trpc } from "@/utils/trpc";
+
+type AilALessonTitles = {
+  sessionId: string;
+  lessonTitle: string;
+  keyStage: string;
+  subject: string;
+  hasThreeCycles: boolean;
+}[];
+
+const LessonFinder = ({ lessons }: { lessons: AilALessonTitles }) => {
+  const [searchText, setSearchText] = useState<string>("");
+  const [owaLessons, setOwaLessons] = useState<OakOpenApiSearchSchema | []>();
+  const { data: owaData } = trpc.additionalMaterials.searchOWALesson.useQuery(
+    { query: searchText },
+    { enabled: !!searchText },
```

Comment:
For production I would think about debouncing while they type, but it's not essential for now